lib/Rabbit.js

```diff
@@ -0,0 +1,253 @@
+                continue;
+            }
+            if ("pubPayload" in this._config) {
+                const content = JSON.stringify(pubPayload);
+                const channel = await connection.createChannel();
+                await channel.assertExchange(exchangeName, "fanout", { durable: true });
+                channel.publish(exchangeName, "", Buffer.from(content), { persistent: true });
+                // note that the queue has already been used for publication
+                this._publishedQueueList.push(exchangeName);
+            }
+        }
+        return true;
+    }
+
+    async _sub(connection) {
+        if (!this._config.sub) {
+            return false;
+        }
+        // extract config
+        const { namespace, serviceName, sub, subNotification, publisher } = this._config;
+        // create channel & listen to event
+        for (const registryName in sub) {
+            for (const eventName in sub[registryName]) {
+                const publisherName = registryName in publisher && eventName in publisher[registryName] ? publisher[registryName][eventName] : "publisher";
+                const callback = sub[registryName][eventName];
+                const exchangeName = `${namespace}.${publisherName}.${eventName}`;
+                const queueName = `${namespace}.${publisherName}.${serviceName}.${eventName}`;
+                const channel = await connection.createChannel();
+                await channel.assertQueue(queueName, {durable: true});
+                await channel.assertExchange(exchangeName, "fanout", {durable: true});
+                await channel.bindQueue(queueName, exchangeName, "");
+                channel.consume(queueName, (message) => {
+                    try {
+                        const payload = JSON.parse(message.content.toString());
+                        callback(payload);
+                        channel.ack(message);
+                    } catch (error) {
+                        channel.nack(message);
+                    }
+                });
+            }
+        }
+        // create notification channel & listen to event
+        for (const eventName in subNotification) {
+            const callback = subNotification[eventName];
+            const queueName = `${namespace}.${serviceName}.${eventName}`;
+            const channel = await connection.createChannel();
+            channel.assertQueue(queueName, {durable: true});
+            channel.consume(queueName, (message) => {
+                try {
+                    const payload = JSON.parse(message.content.toString());
+                    callback(payload);
+                    channel.ack(message);
+                } catch (error) {
+                    channel.nack(message);
+                }
+            });
+        }
+        return true;
+    }
+
+    async _pubQueue(connection) {
+        if (!this._config.pubQueue || !this._connection) {
```

Comment:
if we want this method to be more "isolated", let's use `connection` from params instead of `this._connection` from instance

Reply:
suggestion accepted